src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -0,0 +1,188 @@
+.. ## Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _tpls-label:
+
+*********************
+Third-party Libraries
+*********************
+
+Axom dependencies are grouped into four categories: Git submodules,
+built-in TPLs in the Axom source tree, system-level TPLs, and other 
+TPL libraries. The following sections describe how to install and update 
+these dependencies for Axom.
+
+  * How does one add a new compiler or platform to the mix?
+  * How does one build a new set of TPLs for a single platform or compiler
+    for testing?
+  * What is the procedure for changing versions of one or more TPLs?
+  * How do we keep things straight when using different TPL versions for 
+    different branches?
+  * How to use the scripts for team TPL support vs. local development 
+    and experimentation?
+  * Others?
+
+
+=============
+Updating TPLs
+=============
+
+Git submodules
+--------------
+
+Currently, Axom uses three external packages that appear in the repo
+as Git submodules. These are the following, including the location of the
+package in the Axom source tree:
+
+  * `BLT <https://github.com/LLNL/blt.git>`_, which is the CMake-based build
+    system we use. Location: ``axom/src/cmake/blt``.
+  * `Axom Data <https://github.com/LLNL/axom_data.git>`_, which is where we
+    maintain data files used in testing Axom. Location: ``axom/data``.
+  * `Uberenv <https://github.com/LLNL/uberenv.git>`_, which contains Python
+    scripts we use to help automate building third-party dependencies for
+    development and deployment. Location: ``axom/scripts/uberenv``.
+
+There is no software installation process for these dependencies in the 
+traditional sense. To update one of these packages in Axom, simply go into
+its directory in Axom and check out a new version. If a version is intended
+to be changed in the Axom repo, make the version change on a branch and 
+submit a GitHub pull request as you would do for other software changes.
+
```

Comment:
Should we add a link to our Quick-Start to remind people how to configure/build the code?

E.g. to one of the following:
* https://axom.readthedocs.io/en/docs-white238-tpl_update/docs/sphinx/quickstart_guide/config_build.html
* https://axom.readthedocs.io/en/docs-white238-tpl_update/docs/sphinx/quickstart_guide/index.html